juju/juju
# [JUJU-1790] Watch secret deletes and sync unit state

Unit agents maintain persistent state about obsolete and consumed secret revisions. This PR enhances the existing watchers to report on secret deletions so that deleted secrets can be removed from said state. ie the consumed secrets and obsolete revisions watchers need to track the content and surface removals instead of just embedding a standard collection watcher.

On the unit agent side, the remote state snapshot now includes such deleted secrets. A new no op operation is added to allow the state changes to be saved when the operation is committed.

## Checklist


## QA steps

You need to look at the unitstates collection in mongo. After creating, sharing, and consuming a secret, remove it.
Check that the unitstates record for both the secret owner and secret consumer units has that secret remove from the associated unit state record.
